modules/fedimint-ln-client/src/receive.rs

```diff
@@ -200,14 +193,30 @@ impl LightningReceiveConfirmedInvoice {
     ) -> Result<IncomingContractAccount, LightningReceiveError> {
         let contract_id = (*invoice.payment_hash()).into();
         loop {
+            // Consider time before the api call to account for network delays
+            let now_since_epoch = fedimint_core::time::duration_since_epoch();
             match global_context
                 .module_api()
-                .wait_preimage_decrypted(contract_id)
+                .get_decrypted_preimage_status(contract_id)
                 .await
             {
-                Ok((incoming_contract_account, preimage)) => match preimage {
-                    Some(_) => return Ok(incoming_contract_account),
-                    None => return Err(LightningReceiveError::InvalidPreimage),
+                Ok((incoming_contract_account, status)) => match status {
+                    DecryptedPreimageStatus::Pending => {
+                        // only when we are sure that the invoice is still pending that we can check
+                        // for a timeout
+                        const BUFFER: f64 = 1.10; // 10% buffer to account for clock skew
+                        let invoice_since_epoch_secs =
```

Comment:
**issue:** 

```
> date +%s
1702448004
```

Isn't 10% of `1702448004`s ... 5 years? Do we really need such a buffer? Seems like 60s should be enough for everyone?

BTW. "buffer" seems like a wrong word. Would just inline it to not have to come up with a better name.

Reply:
The previous implementation applied it on top of the relative timeout. So 6min per hour.

Reply:
@douglaz the factor should be applied on the relative expiry time, not the absolute one.

Reply:
I'm sorry, my commit didn't include the fix. Look now.

Reply:
I think fixed offset between computers is much more likely than clocks going 10% slower/faster (which would right away caused fixed offset anyway). Unless I'm missing something.

BTW. Makes me wonder if federation should have a consensus on time as well. :D

Reply:
Also good to always allow a fixed grace period as long as there is one.